PyStemmusScope/stemmus_scope.py

```diff
@@ -94,23 +155,34 @@ def run(self) -> str:
         Args:
 
         Returns:
-            Tuple with stdout and stderr
+            string, the model log
         """
-
-        # run the model
-        args = [f"{self.exe_file} {self.cfg_file}"]
-        result = subprocess.run(
-            args, stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True, check=True,
-        )
-        stdout = result.stdout
-
-        # TODO return log info line by line!
-        logger.info("%s", stdout)
-
-        return stdout
+        if self.exe_file:
+            # run using MCR
+            args = [f"{self.exe_file} {self.cfg_file}"]
+            # set matlab log dir
+            os.environ['MATLAB_LOG_DIR'] = str(self._config["InputPath"])
+            result = _run_sub_process(args, None)
+        if self.sub_process=="Matlab":
+            # set Matlab arguments
+            path_to_config = f"'{self.cfg_file}'"
+            command_line = f'matlab -r "STEMMUS_SCOPE_exe({path_to_config});exit;"'
+            args = [command_line, "-nodisplay", "-nosplash", "-nodesktop"]
```

Comment:
```suggestion
            eval_code= f"STEMMUS_SCOPE_exe({path_to_config});exit;"
            args = ["matlab", "-r", eval_code, "-nodisplay", "-nosplash", "-nodesktop"]
```
This change is probably also required for the Matlab, for Windows compatibility. See the suggestion on lines 179-180.

Reply:
I commit your suggestion here, but I will check if the code works on linux too.

Reply:
As discussed the separate args don't work on Linux. I used `shlex.join()` for it.